### PR TITLE
Don't always create ErrorMessage in ProcessData::FindModuleByAddress

### PR DESCRIPTION
**Pull request by dpallotti**
This speeds up `CreateTopDownViewFromPostProcessedSamplingData` by ~40% and
`CreateBottomUpViewFromPostProcessedSamplingData` by ~30%! (as per the timings
we print in the logs with a capture with 4M samples).

### CODE REVIEW

File: src/ClientData/ProcessData.cpp [hunk 1]
```diff
@@ -135,17 +135,21 @@ ErrorMessageOr<ModuleInMemory> ProcessData::FindModuleByAddress(uint64_t absolut
...
 
-  ErrorMessage not_found_error = ErrorMessage(absl::StrFormat(
-      "Unable to find module for address %016x: No module loaded at this address by process %s",
-      absolute_address, process_info_.name()));
-
   auto it = start_address_to_module_in_memory_.upper_bound(absolute_address);
-  if (it == start_address_to_module_in_memory_.begin()) return not_found_error;
+  if (it == start_address_to_module_in_memory_.begin()) {
+    return ErrorMessage{absl::StrFormat(
+        "Unable to find module for address %016x: No module loaded at this address by process %s",
```
Comment (florian-kuebler) [inline src/ClientData/ProcessData.cpp:141]: Can we still extract that template string as a constant?

**Review by florian-kuebler — APPROVED** [review]
Other than that, LGTM. Thanks!

**Review by antonrohr — APPROVED** [review]
LGTM!
Thank you for finding these performance issues

File: src/ClientData/ProcessData.cpp (hunk 1 above)
Comment (dpallotti) [reply src/ClientData/ProcessData.cpp]: I had thought about it but I was undecided. Good to have an opinion. Done.

**Pull request merged**